internal/install/_static/terraform_deployer_run.sh

```diff
@@ -16,6 +16,15 @@ cleanup() {
 }
 trap cleanup EXIT INT TERM
 
+# Save GCP credentials on disk and perform authentication
+# NOTE: this is required for bq (and maybe other gcloud related tools) to authenticate
+export "GOOGLE_APPLICATION_CREDENTIALS=/root/.config/gcloud/application_default_credentials.json"
+echo "$GOOGLE_CREDENTIALS" > "$GOOGLE_APPLICATION_CREDENTIALS"
+gcloud auth login --cred-file "$GOOGLE_APPLICATION_CREDENTIALS"
```

Comment:
If this is required by `bq`, maybe we can execute it just before running `bq`?

Reply:
In case we are going to use other tools, something possible given the new Docker image, I think it make sense to unify authentication so all tools leverage the same credentials and should not do it multiple times (with possible concurrency risks). 
Unifying authentication also prevents that Terraform and tools run with different credentials, something that would make the CI harder to debug.

Reply:
I reviewed the pro and cons of moving authentication where `bq` is used. The main cons I see is around how this affects authentication globally. `gcloud auth` state is globally shared across all instances of `gcloud` (and related tools). So doing that within terraform code would create side effects that may be unintuitive or difficult to debug.

What is the downside of the current approach? (given that it does not trigger always but only when it makes sense)

Reply:
> What is the downside of the current approach? (given that it does not trigger always but only when it makes sense)

I think you included the answer here. If there is a way of not introducing the coupling on auth, why should we do this? Please remember that it isn't a dedicated GCP service.

> gcloud auth state is globally shared across all instances of gcloud (and related tools). So doing that within terraform code would create side effects that may be unintuitive or difficult to debug.

It means that you can call it this way:

````
gcloud auth login
bq
gcloud auth revoke
```

Reply:
You are at risk of race conditions if multiple resources are using this commands. Terraform runs in parallel and this command affects a global state.

> If there is a way of not introducing the coupling on auth, why should we do this?

My reasoning about this is that coupling is already present as in "terraform requires credentials to run", so if I use it with GCP resources is coupled with GCP auth, if I use it with AWS is coupled with AWS auth. How could we avoid this? I agree that ideally credentials would be passed in env vars and we should not rely on other tools.

> Please remember that it isn't a dedicated GCP service.

Could we consider having a dedicated deployer for gcp? Would this help untangling this?